### ISSUE TITLE
Improve the functional programming of the stellar implementations.

### ISSUE THREAD

**Issue by SanGeeky** [kommitters/mintacoin#64]
ℹ️ This issue is part of Epic #55

## Objective

Currently, we have some stellar implementations in the modules located in `lib/mintacoin/*/crypto/stellar.ex` with the Stellar SDK, but we need to improve the functional programming in these modules.

For example: in the `lib/mintacoin/accounts/crypto/stellar.ex` we have the `create_account/1` function that contains multiple functions and calls that can be separated by functionality, in order to improve the readability and maintainability of the code.

This aims to make the code follow a functional pattern and approach the capabilities that Elixir provides.

## Changes
1. Update account keypair statement
    ```elixir
    # lib/mintacoin/accounts/crypto/stellar.ex#19
    {fund_public_key, _fund_private_key} = fund_keypair = fund_key_pair_from_system()
    ```
    Create a function called `funder_account_information/0` that only returns the `public_key` and `signature`:
    ```elxiir
    %{ funder_public_key: pk,  funder_signature: signature  }
    ```

2. Move the line where is defined the Create Account operation:
    ```elixir
    # lib/mintacoin/accounts/crypto/stellar.ex#26
    operation =
          TxBuild.CreateAccount.new(
            destination: public_key,
            starting_balance: 1
          )
    ```

    The function should be called `build_create_account_operation/2` and the parameters of the function must contain the `public_key` and the `starting_balance`.

3. Move the following parts of the code to a function called `build_envelope/3` that contains the following operations
    ```elixir
    # lib/mintacoin/accounts/crypto/stellar.ex#22
    source_account = TxBuild.Account.new(fund_public_key)
    {:ok, seq_num} = Horizon.Accounts.fetch_next_sequence_number(fund_public_key)
    sequence_number = TxBuild.SequenceNumber.new(seq_num) 
    
    # lib/mintacoin/accounts/crypto/stellar.ex#34
    {:ok, envelope} =
          source_account
          |> TxBuild.new(sequence_number: sequence_number)
          |> TxBuild.add_operation(operation)
          |> TxBuild.sign(signature)
          |> TxBuild.envelope()
    ```

4. Refactor the transaction submitting to a function called `execute_transaction/1`
    ```elixir
    # lib/mintacoin/accounts/crypto/stellar.ex#41
    envelope
    |> Horizon.Transactions.create()
    |> format_response(public_key, secret_key)
    ```

> Keep in mind to chain all the previous functions using the pipe operator.

**Comment by Odraxs** [comment]
Hi, I would like to be assigned to this issue.

**Comment by SanGeeky** [comment]
@Odraxs I'll assign this issue to you, do not forget to keep up to date with your branch with `v0.3`.

PD, I just updated the issue description with a minimum specification

**Comment by Odraxs** [comment]
hello @SanGeeky , I have a doubt about refactor #4 the function arity is /1 but the secret and public keys are needed, in stellar.ex#20 are delcared and the public_key value is needed for `build_create_account_operation/2` as well

I guess `execute_transaction/1` has to be arity 3 because the public_key has to be the same ?

**Comment by SanGeeky** [comment]
@Odraxs you're right `execute_transaction/3` required the `public_key` and `secret_key` for the wallet creation, and as you mentioned this keypairs are created with `{public_key, secret_key} = KeyPair.random()`.

**Comment by SanGeeky** [comment]
Closed by #83.
